File: test_doubles/csharp/TestDoubles/TestDoubles/Repository/PurchaseRepository.cs
using System;
using System.Collections.Generic;
using System.Linq;
using System.Text;
using TestDoubles.Domain;

namespace TestDoubles.Repository
{
    public class PurchaseRepository
    {
        public Purchase CreatePurchase(Customer customer)
        {
            // Here a new purchase would be created in the database
            return new Purchase();
        }

        public void AddProduct(Purchase purchase, Product product)
        {
            
        }

        public Invoice CreateInvoice(Purchase purchase)
        {
            return new Invoice();
        }

        public Invoice CreateInvoice(Customer customer, IList<Product> products)
        {
            throw new NotImplementedException();
        }
    }
}

File: test_doubles/java/TestDoubles/src/main/java/repository/PurchaseRepository.java
package repository;

import domain.Customer;
import domain.Invoice;
import domain.Product;
import domain.Purchase;

import java.util.List;

public class PurchaseRepository {
    public Purchase createPurchase(Customer customer) {
        // Here a new purchase would be created in the database
        return new Purchase();
    }

    public void addProduct(Purchase purchase, Product product) {

    }

    public Invoice createInvoice(Purchase purchase) {
        return new Invoice();
    }

    public Invoice createInvoice(Customer customer, List<Product> products) {
        throw new UnsupportedOperationException("Implement this!");
    }

}
